addons/sale/tests/test_sale_order.py

```diff
@@ -339,6 +341,65 @@ def test_order_auto_lock_with_public_user(self):
         self.assertFalse(public_user.has_group('sale.group_auto_done_setting'))
         self.assertTrue(self.sale_order.locked)
 
+    def test_order_status_email_is_sent_synchronously_if_not_configured(self):
+        """ Test that the order status email is sent synchronously when nothing is configured. """
+        self.env['ir.config_parameter'].set_param('sale.async_emails', 'False')
+
+        self.sale_order._send_order_notification_mail(self.confirmation_email_template)
+        self.assertFalse(
+            self.env['ir.cron.trigger'].search_count([('cron_id', '=', self.async_emails_cron.id)]),
+            msg="The email should be sent synchronously when the system parameter is not set.",
+        )
+
+    def test_order_status_email_is_sent_asynchronously_if_configured(self):
+        """ Test that the order status email is sent asynchronously when configured. """
+        self.env['ir.config_parameter'].set_param('sale.async_emails', 'True')
+
+        self.sale_order._send_order_notification_mail(self.confirmation_email_template)
+        self.assertTrue(
+            self.sale_order.pending_email_template_id,
+            msg="The email template should be saved on the sales order.",
+        )
+        self.assertTrue(
+            self.env['ir.cron.trigger'].search_count([('cron_id', '=', self.async_emails_cron.id)]),
+            msg="The asynchronous email sending cron should be triggered.",
+        )
+
+    def test_async_emails_cron_does_not_trigger_itself(self):
+        """ Test that the asynchronous email sending cron does not loop indefinitely. """
+        self.env['ir.config_parameter'].set_param('sale.async_emails', 'True')
+        self.sale_order.pending_email_template_id = self.confirmation_email_template
+
+        with self.enter_registry_test_mode():
+            self.env.ref('sale.send_pending_emails_cron').method_direct_trigger()
+        self.assertFalse(
+            self.sale_order.pending_email_template_id,
+            msg="The email template should be removed from the sales order.",
+        )
+        self.assertFalse(
+            self.env['ir.cron.trigger'].search_count([('cron_id', '=', self.async_emails_cron.id)]),
+            msg="The email should be sent synchronously when requested by the cron.",
+        )
+
+    def test_draft_quotation_followers(self):
+        sale_order = self.env['sale.order'].create({
+            'partner_id': self.partner1.id,
+        })
+
+        sale_order.partner_id = self.partner2
+
+        self.assertNotIn(self.partner2, sale_order.message_partner_ids)
+
+    def test_sent_quotation_followers(self):
+        sale_order = self.env['sale.order'].create({
+            'partner_id': self.partner1.id,
+        })
+        sale_order.action_quotation_sent()
+
+        sale_order.partner_id = self.partner2
+
+        self.assertIn(self.partner2, sale_order.message_partner_ids)
+
```

Comment:
Those two tests were removed recently by TDE, seems like you brought them back on rebase :eyes: